### PR TITLE
doc/telemetry: add exported events schema

### PR DESCRIPTION
**Pull request by bobheadxi**
This adds a generated markdown reference (thanks @burmudar!) that can serve as a source-of-truth for exactly what data we do and don't export from Sourcegraph instances.

## Test plan

```
bazel run //doc/dev/background-information/telemetry:write_telemetrygateway_doc
```

Preview of what that looks like: https://github.com/sourcegraph/sourcegraph/blob/09-27-telemetry_generate_export_schema/doc/dev/background-information/telemetry/protocol.md

Output is a little funky, unless https://github.com/pseudomuto/protoc-gen-doc/pull/520 gets landed one day (for that new `br` template function)

## Preview 🤩
[Preview Link](https://docs.sourcegraph.com/@09-27-telemetry_generate_export_schema)

### CODE REVIEW

**Comment by bobheadxi** [comment]
Current dependencies on/for this PR:
* main
  * **PR #57104** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/57104" target="_blank">Graphite</a> 
    * **PR #57106** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/57106" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/57106?utm_source=stack-comment).

File: go.mod [hunk 1]
```diff
@@ -233,18 +233,18 @@ require (
...
 	go.uber.org/zap v1.26.0
-	golang.org/x/crypto v0.12.0
-	golang.org/x/net v0.14.0
+	golang.org/x/crypto v0.13.0
+	golang.org/x/net v0.15.0
 	golang.org/x/oauth2 v0.10.0
 	golang.org/x/sync v0.3.0
 	golang.org/x/sys v0.12.0
 	golang.org/x/time v0.3.0
 	// golang.org/x/tools should not be upgraded until https://github.com/bazelbuild/bazel-gazelle/issues/1579
```
Comment (burmudar) [inline go.mod:242]:
```suggestion
	// golang.org/x/tools should not be upgraded until https://github.com/bazelbuild/rules_go/issues/3640
```

Comment (burmudar) [reply go.mod:242]: Old issue got closed

File: go.mod [hunk 2]
```diff
@@ -233,18 +233,18 @@ require (
...
 	// is resolved.
-	golang.org/x/tools v0.10.0
+	golang.org/x/tools v0.13.0
```
Comment (burmudar) [inline go.mod:244]:
```suggestion
	golang.org/x/tools v0.10.0
```

Comment (burmudar) [reply go.mod:244]: 0.13 breaks `nogo` ... the bazel go linters.

Reminder to run `bazel run :gazelle-update-repos` when you've made the above change

File: go.mod (hunk 1 above)
Comment (bobheadxi) [reply go.mod]: "problem solved" 😎

File: internal/telemetrygateway/v1/protoc-gen-doc.tmpl [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+{{range .Methods -}}
+  | {{.Name}} | [{{.RequestLongType}}](#{{.RequestFullType | anchor}}){{if .RequestStreaming}} stream{{end}} | [{{.ResponseLongType}}](#{{.ResponseFullType | anchor}}){{if .ResponseStreaming}} stream{{end}} | {{p .Description}} |
+{{end}}
+{{end}} <!-- end services -->
+
+{{end}}
+
+## Scalar Value Types
+
+| .proto Type | Notes | C++ | Java | Python | Go | C# | PHP | Ruby |
```
Comment (unknwon) [inline internal/telemetrygateway/v1/protoc-gen-doc.tmpl:107]: 😂 do we need this many languages displayed?

**Review by unknwon — APPROVED** [review]

File: internal/telemetrygateway/v1/protoc-gen-doc.tmpl (hunk 3 above)
Comment (bobheadxi) [reply internal/telemetrygateway/v1/protoc-gen-doc.tmpl]: Good point, got rid of the language-specific stuff from the template 😆

**Pull request merged**